### PR TITLE
Type checkers part 2

### PR DESCRIPTION
**Pull request by topepo**
Complement of #1387

This _should_ do it for the new type checkers in this package.

### CODE REVIEW

File: R/other.R [hunk 1]
```diff
@@ -107,13 +107,7 @@ step_other <-
            objects = NULL,
            skip = FALSE,
            id = rand_id("other")) {
-    if (!is_tune(threshold)) {
```
Comment (topepo) [inline R/other.R:110]: There's a lot of moving things here and in #1387. My rationale is that, if a the tuning process inserts a value, we would not check it. For example, `finalize_recipe()` does use the constructor. 

It does defer the checking until prep-time which is later than this but it seems like the only way to really be sure that we are getting corrrect inputs.

File: tests/testthat/test-window.R [hunk 2]
```diff
@@ -14,35 +14,39 @@ rec <- recipe(~., data = sim_dat)
...
-    rec %>% step_window(y1, size = 2)
+    rec %>% step_window(y1, size = 2) %>% prep()
   )
   expect_snapshot(error = TRUE,
-    rec %>% step_window(y1, size = -1)
+    rec %>% step_window(y1, size = -1) %>% prep()
   )
   expect_snapshot(
-    rec %>% step_window(y1, size = pi)
+    rec %>% step_window(y1, size = 3 + .Machine$double.eps) %>% prep()
```
Comment (topepo) [inline tests/testthat/test-window.R:39]: So I don't know that we need both of the tests using `.Machine$double.eps` but it did affirm by beleif of what _should_ happen given how `all_equal()` works.

File: R/YeoJohnson.R [hunk 3]
```diff
@@ -109,6 +109,14 @@ step_YeoJohnson_new <-
 prep.step_YeoJohnson <- function(x, training, info = NULL, ...) {
   col_names <- recipes_eval_select(x$terms, training, info)
   check_type(training[, col_names], types = c("double", "integer"))
+  check_number_whole(x$num_unique, args = "num_unique")
+  check_bool(x$na_rm, arg = "na_rm")
+  if (!is.numeric(x$limits) | any(is.na(x$limits)) | length(x$limits) != 2) {
```
Comment (EmilHvitfeldt) [inline R/YeoJohnson.R:114]:
```suggestion
  if (!is.numeric(x$limits) || any(is.na(x$limits)) || length(x$limits) != 2) {
```

**Review by EmilHvitfeldt — APPROVED** [review]
amazing!

**Pull request merged**